File: Silver/binarysearch/binarysearch.cpp
#include <iostream>
#include <vector>

using namespace std;

int binarySearch(const vector<int> &a, int target)
{
    int l, r;
    l = 0;
    r = a.size() - 1;
    int mid;
    while (l < r)
    {
        mid = l+ (r - l) / 2;
        if (a[mid] == target)
        {
            return mid;
        }
        else if (a[mid] > target)
        {
            r = mid - 1;
        }
        else
        {
            l = mid + 1;
        }
    }
    return -1;
}

int main()
{
    vector<int> arr = {1, 2, 4, 7, 10, 50, 112, 1000, 3000, 4661};

    int target = 10;

    cout << binarySearch(arr, target) << endl;
}

File: Silver/binarysearch/binarysearch.py
def binary_search(array: list[int], target: int) -> int:
    left: int = 0
    right: int = len(array)-1
    while(left<right):
        mid: int = left+(right-left)//2
        if array[mid] == target:
            return mid
        if array[mid] > target:
            right = mid-1
        else:
            left = mid+1
    return -1

def main():
    array: list[int] = [1, 2, 4, 5, 7, 11, 41, 100]
    target: int = 11
    print(binary_search(array, 11))

if __name__ == '__main__':
    main()